array/js/array.js

```diff
@@ -0,0 +1,92 @@
+function isArray(object) {
+  return Array.isArray(object);
+}
+
+function isArray2(object) {
+  return object.constructor.name == "Array";
+}
+
+function isArray3(object) {
+  return {}.toString.call(object) == "[object Array]";
+}
+
+function range(start, end, step) {
+  if (end == undefined) {
+    end = start;
+    start = 1;
+  }
+  if (step == undefined) {
+    step = 1;
+  }
+  var array = [];
+  for (var i = start; i < end + 1; i = i + step) {
+    array.push(i);
+  }
+  return array;
+}
+
+function compact(array) {
+  return array.filter(function (element) {
+    return element != false;
+  });
+}
+
+function sum(array) {
+  return array.reduce(function (sum, element) {
+    return sum + element;
+  }, 0)
+}
+
+function sum2(array) {
+  var sum = 0;
+  for (var i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
+function unique(array) {
+  var set = {};
+  return array.filter(function (element) {
+    var isExist = set[element];
```

Comment:
it won't work with objects
remember what type has index: String. when you do `set[element]`, toString() is called on element